cylc/flow/scheduler_cli.py

```diff
@@ -565,15 +566,16 @@ def _version_check(
             ))
             if is_terminal():
                 # we are in interactive mode, ask the user if this is ok
-                options.upgrade = prompt(
-                    cparse(
-                        'Are you sure you want to upgrade from'
-                        f' <yellow>{last_run_version}</yellow>'
-                        f' to <green>{__version__}</green>?'
-                    ),
-                    {'y': True, 'n': False},
-                    process=str.lower,
-                )
+                with handle_sigint():
```

Comment:
Why not use:

```python
with suppress(KeyboardInterrupt):
```

Or:


```python
try:
    ...
except KeyboardInterrupt:
    sys.exit(0)
```

(As appropriate)

Reply:
I think that requires Ctrl+C twice while you are inside of the `input()` loop (or maybe doesn't work at all, can't remember), whereas this handles Ctrl+C immediately